### PR TITLE
vec_order() works recursively to handle data frame columns.

### LINKED ISSUE #507: vec_order() with a data frame column, not enough spicing
``` r
library(vctrs)
library(tibble)

df <- tibble(x = 1:5, d = tibble(y = 1:5, z = 1:5, a = letters[1:5]))
vec_order(df)
#> Must use a vector in `[`, not an object of class matrix.
```

it's happening at the `exec` line in `order_proxy()` 

```r
    args <- unname(proxy)
    exec("order", !!!args, decreasing = decreasing, na.last = na.last)
```

because it's "not spliced enough" : 

``` r
library(vctrs)
library(tibble)

df <- tibble(x = 1:5, d = tibble(y = 1:5, z = 1:5, a = letters[1:5]))
proxy <- vec_proxy_compare(df)
  
order(proxy[[1]], proxy[[2]])
#> Must use a vector in `[`, not an object of class matrix.

# further spliced
order(proxy[[1]], proxy[[2]][[1]], proxy[[2]][[2]], proxy[[2]][[3]])
#> [1] 1 2 3 4 5
```

### PR DESCRIPTION
**Pull request by romainfrancois**
closes #507

``` r
library(vctrs)
library(tibble)

df <- tibble(x = 1:5, d = tibble(y = 1:5, z = 1:5, a = letters[1:5]))
vec_order(df)
#> [1] 1 2 3 4 5

# follow up to https://github.com/r-lib/vctrs/pull/506
df <- data.frame(times = 1:5, x = 1:5)
df$times <- as.POSIXlt(seq.Date(Sys.Date(), length.out = 5, by = "day"))
vec_order(df)
#> [1] 1 2 3 4 5
```

### CODE REVIEW

File: R/compare.R [hunk 1]
```diff
@@ -139,7 +139,12 @@ order_proxy <- function(proxy, direction = "asc", na_value = "largest") {
     if (vec_size(proxy) == 0L) {
       return(integer(0L))
     }
-    args <- unname(proxy)
+    args <- map(unname(proxy), function(.x) {
+      if (is.data.frame(.x))
+        vec_order(.x, direction = direction, na_value = na_value)
```
Comment (lionel-) [inline R/compare.R]: We wrap branch expressions in curly blocks, as per the style guide.

File: R/compare.R [hunk 2]
```diff
@@ -140,10 +140,10 @@ order_proxy <- function(proxy, direction = "asc", na_value = "largest") {
...
     }
     args <- map(unname(proxy), function(.x) {
-      if (is.data.frame(.x))
-        vec_order(.x, direction = direction, na_value = na_value)
-      else
-        .x
+      if (is.data.frame(.x)) {
+        .x <- vec_order(.x, direction = direction, na_value = na_value)
+      }
+      .x
```
Comment (lionel-) [inline R/compare.R]: There's an `else` missing right?

Comment (romainfrancois) [reply R/compare.R]: it used to be `if(is.data.frame(.x)) vec_order(.x, ...) else .x` now `.x` get reassigned and then we always return `.x` I can rollback to previous version but it felt weird to have `{ .x }`

**Comment by lionel-** [comment]
Thanks!

**Pull request merged**